app/models/reservation.rb

```diff
@@ -1,5 +1,20 @@
 class Reservation < ApplicationRecord
   belongs_to :user
-  has_many :equipment_reservations, dependent: :destroy
-  has_many :equipments, through: :equipment_reservations
+  belongs_to :equipment
+
+  validate :date_available?
+
+  private
+
+  def date_available?
+    equipment&.dates_reserved&.each do |date|
+      date_datetime = date.to_datetime
+      reserved_datetime = reserved_date.to_datetime
+      reserved_date_plus_duration = reserved_datetime + equipment.duration.days
+
+      if date_datetime >= reserved_datetime && date_datetime < reserved_date_plus_duration
+        errors.add(:reserved_date, 'is already taken')
+      end
+    end
+  end
```

Comment:
Great job with this validation! 

I think we should also add some validations of presence in the other attributes. Let me handle that too